On low end devices with sparse amount of cpus, cryptographic hash functions
are tad overkill for calculating flow hashes, and take a lot of cpu time too.

--- a/net/core/flow_dissector.c
+++ b/net/core/flow_dissector.c
@@ -1779,8 +1779,7 @@ static inline u32 __flow_hash_from_keys(
 
 	__flow_hash_consistentify(keys);
 
-	hash = siphash(flow_keys_hash_start(keys),
-		       flow_keys_hash_length(keys), keyval);
+	hash = jhash(flow_keys_hash_start(keys), flow_keys_hash_length(keys), *(u32*)keyval);
 	if (!hash)
 		hash = 1;
 
